Paragraph on Market Dynamics (Part II) (#160)
celo-org/governance: CGPs/cgp-0060.md

</p>
*depending on how the pool is utilized, governance might add more liquidity in the future via additional governance proposals until GigaMento comes into effect, see [Rebalancing](#rebalancing-16).

## Market Dynamics and Reserve Balance

## Arbitrage and DEX Competition
The setup of the StableSwap pool is competitive. The transactions fee is on a similar level as for stable pairs on other DEXs and additionally it uses a constant sum market making function. As a consequence, especially larger transactions will be cheaper with the constant sum StableSwap pool.

Arbitrageurs will take advantage of the arbitrage opportunity. In the current market situation (cUSD/USDC price at ~0.995 ), arbitrageurs will buy cUSD below 1 USD in the market and sell it to the StableSwap pool for a cUSD/USDC price of 1. Ignoring fees and slippage, the maximal profit they can earn if cUSD/USD remains at 1 is

$$ 5\,000\,000 (1-0.995) = 25\,000\mathrm{\ USDCet} \ .$$

## Reserve Balance

> The reserve will always be able to unwind
> irrespective of whether arbitrageurs take arbitrage opportunities or not. Under the assumption that the USDC/USD and cUSD/USD prices stay a 1, the net reserve balance (reserve collateral minus debt) will not change.

Reviewer: ```suggestion
As long as USDC/USD and cUSD/USD stay a 1, the reserve will always be able to unwind without a net loss - irrespective of whether arbitrageurs take arbitrage opportunities or not. This is because exchanges happen 1:1 and therefore changes in reserve debt and changes in reserve collateral cancel each other out.
```

</p>
*depending on how the pool is utilized, governance might add more liquidity in the future via additional governance proposals until GigaMento comes into effect, see [Rebalancing](#rebalancing-16).

## Market Dynamics and Reserve Balance

## Arbitrage and DEX Competition
The setup of the StableSwap pool is competitive. The transactions fee is on a similar level as for stable pairs on other DEXs and additionally it uses a constant sum market making function. As a consequence, especially larger transactions will be cheaper with the constant sum StableSwap pool.

> Arbitrageurs will take advantage of the arbitrage opportunity. In the current market situation (cUSD/USDC price at ~0.995 ), arbitrageurs will buy cUSD below 1 USD in the market and sell it to the StableSwap pool for a cUSD/USDC price of 1. Ignoring fees and slippage, the maximal profit they can earn if the cUSD/USD price remains at 1  is

Reviewer: ```suggestion
Arbitrageurs will take advantage of the arbitrage opportunity. In the current market situation (cUSD/USDC price at ~0.995 ), arbitrageurs will buy cUSD below 1 USD in the market and sell it to the StableSwap pool for a cUSD/USDC price of 1. Ignoring fees and slippage, the maximal profit they can earn if cUSD/USD remains at 1 is
```

$$ 5\,000\,000 (1-0.995) = 25\,000\mathrm{\ USDCet} \ .$$

## Reserve Balance
The reserve will always be able to unwind

> 
> $$ \lambda \mathrm{USDCet} - (1-\lambda) \mathrm{cUSD}\ , \ \lambda\in[0,1],$$
> 

Reviewer: ```suggestion
```

</p>
*depending on how the pool is utilized, governance might add more liquidity in the future via additional governance proposals until GigaMento comes into effect, see [Rebalancing](#rebalancing-16).

## Market Dynamics and Reserve Balance

## Arbitrage and DEX Competition

> The setup of the StableSwap pool is competitive. The transactions fee us on a similar level as for stable pairs on other DEXs, and additionally it uses a constant sum market making function. As a consequence, especially larger transactions will be cheaper with the constant sum StableSwap Pool.

Reviewer: ```suggestion
The setup of the StableSwap pool is competitive. The transactions fee is on a similar level as for stable pairs on other DEXs and additionally it uses a constant sum market making function. As a consequence, especially larger transactions will be cheaper with the constant sum StableSwap pool.
```

> Arbitrageurs will take advantage of the arbitrage opportunity. In the current market situation (cUSD/USDC price at ~0.995 ), arbitrageurs will buy cheaper cUSD in the market and sell it at the StableSwap Pool for a cUSD/USDC price of 1. Ignoring fees and slippage, the maximal profit they can make is

Reviewer: ```suggestion
Arbitrageurs will take advantage of the arbitrage opportunity. In the current market situation (cUSD/USDC price at ~0.995 ), arbitrageurs will buy cUSD below 1 USD in the market and sell it to the StableSwap pool for a cUSD/USDC price of 1. Ignoring fees and slippage, the maximal profit they can earn if the cUSD/USD price remains at 1  is
```

$$ 5\,000\,000 (1-0.995) = 25\,000\mathrm{\ USDCet} \ .$$

## Reserve Balance
The reserve will always be able to unwind

$$ \lambda \mathrm{USDCet} - (1-\lambda) \mathrm{cUSD}\ , \ \lambda\in[0,1],$$

> independently whether arbitrageurs take arbitrage opportunities. Under the assumption that the USDC/USD price is 1 and that cUSD/USD around, the reserve balance will not change:

Reviewer: ```suggestion
irrespective of whether arbitrageurs take arbitrage opportunities or not. Under the assumption that the USDC/USD and cUSD/USD prices stay a 1, the reserve balance will not change:
```

> **Case $\lambda =1$:** All 5m pre-minted cUSD have been minted into the market and the collateral of 5m USDCet is in the USDCet pool. Half of the USDCet have been reserve owned already. The additional 5m USDCet worth 5m USD serve as collateral of the 5m pre-minted cUSD that have been minted into the market. In total the unwinding of the pool the reserve balance increases by 5m USDCet collateral for 5mm cUSD which does not change the net reserve value.  

Reviewer: ```suggestion
**Case $\lambda =1$:** All 5m pre-minted cUSD have been distributed to the market and the collateral of 5m USDCet has been added to the USDCet pool. Half of the 10m USDCet in the pool have been reserve owned already. The additional 5m USDCet serve as reserve collateral of the 5m pre-minted cUSD that have been distributed to the market. 
```

> **Case $\lambda =0$:** Additional 5m cUSD have been 'contracted' and collateral of 5m USDCet have been liquidated. 10m cUSD will be burnt. Half of the cUSD have only pre-minted and have not created 'debt'. The additional 5m cUSD remove 'debt' worth 5m USD and release 5m USDC worth 5m USD. In total the unwinding of the pool the reserve balance decreases by 5m USDCet  collateral and 5m cUSD of 'debts' which does not change the net reserve value.  

Reviewer: ```suggestion
**Case $\lambda =0$:** Additional 5m cUSD have been 'contracted' and collateral of 5m USDCet has been paid out in the process. The pool now contains 10m cUSD which will be burnt after unwinding of the pool. 
```

> The case $\lambda <0.5$, i.e. more than 5m cUSD and less than 5m  USDCet, is similar to $\lambda=0$ and the case $\lambda >0.5$, i.e. less than 5m cUSD and more than 5m USDCet, is similar to $\lambda=1$. The only difference is that there is a remaining amount of remaining amount of pre-owned USDCet that will be returned or an remaining amount of preminted cUSD that will be burnt, respectively.

Reviewer: ```suggestion
The case $\lambda <0.5$, i.e. more than 5m cUSD and less than 5m  USDCet, is similar to $\lambda=0$ and the case $\lambda >0.5$, i.e. less than 5m cUSD and more than 5m USDCet, is similar to $\lambda=1$. The only difference is that there is a remaining amount of pre-owned USDCet that will be returned or an remaining amount of preminted cUSD that will be burnt, respectively.
```

> 
> **Case $\lambda =1$:** All 5m pre-minted cUSD have been minted into the market and the collateral of 5m USDCet is in the USDCet pool. Half of the USDCet have been reserve owned already. The additional 5m USDCet worth 5m USD serve as collateral of the 5m pre-minted cUSD that have been minted into the market. In total the unwinding of the pool the reserve balance increases by 5m USDCet collateral for 5mm cUSD which does not change the net reserve value.  
> **Case $\lambda =0$:** Additional 5m cUSD have been 'contracted' and collateral of 5m USDCet have been liquidated. 10m cUSD will be burnt. Half of the cUSD have only pre-minted and have not created 'debt'. The additional 5m cUSD remove 'debt' worth 5m USD and release 5m USDC worth 5m USD. In total the unwinding of the pool the reserve balance decreases by 5m USDCet  collateral and 5m cUSD of 'debts' which does not change the net reserve value.  
> The case $\lambda <0.5$, i.e. more than 5m cUSD and less than 5m  USDCet, is similar to $\lambda=0$ and the case $\lambda >0.5$, i.e. less than 5m cUSD and more than 5m USDCet, is similar to $\lambda=1$. The only difference is that there is a remaining amount of remaining amount of pre-owned USDCet that will be returned or an remaining amount of preminted cUSD that will be burnt, respectively.
> 
>  
> 

Reviewer: ```suggestion
```

The cUSD/USDCet pool will offer cUSD (and USDCet) at an incorrect price that is either higher (cUSD/USDCet < 1) or lower (cUSD/USDCet > 1) than the market price. The availability of expensive/cheap liquidity can have a price impact that is undesirable for cUSD as it is purely caused by a USDCet price deviation.

 
## Market Interference

> Whenever arbitrageurs have matched the StableSwap cUSD/USDCe price with the price on other pools, most users will trade with the reserve StableSwap pool since it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shifts trading volume towards the StableSwap pool.

Reviewer: ```suggestion
Whenever arbitrageurs have matched the the prices on other pools with the StableSwap cUSD/USDCe price, most users will trade with the reserve StableSwap pool since it has lower fees and slippage especially for larger trades.
```

> Whenever arbitrageurs have matched cUSD/USDCe prices on other pools with the StableSwap pool, most users will trade with the reserve StableSwap pool. Due to the setup (see [Arbitrage and Dex Competition](#arbitrage-and-dex-competition) ) it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shift of trading volume towards the StableSwap pool will become even more likely.

Reviewer: ```suggestion
Whenever arbitrageurs have matched the StableSwap cUSD/USDCe price with the price on other pools, most users will trade with the reserve StableSwap pool since it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shifts trading volume towards the StableSwap.
```

> In other words DEXs have to compete with reserve liquidity. This is a dynamic which we would like to avoid in general. In the current situation we believe that the benefit for the entire Celo ecosystem outweighs the risk of a competitive market intervention.

Reviewer: ```suggestion
In other words, cUSD/USDCet pools on other DEXs will have to compete with the reserve liquidity provided via this CGP. This is a dynamic which we would like to avoid in general. In the current situation we however believe that the benefit of governance-owned cUSD/USDCet liquidity for the entire Celo ecosystem outweighs the drawbacks.
```

> Whenever arbitrageurs have matched the StableSwap cUSD/USDCe price with the price on other pools, most users will trade with the reserve StableSwap pool since it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shifts trading volume towards the StableSwap.
> In other words DEXs have to compete with reserve liquidity. This is a dynamic which we would like to avoid in general. In the current situation we believe that the benefit for the entire Celo ecosystem outweighs the risk of a competitive market intervention.

Reviewer: ```suggestion
Whenever arbitrageurs have matched the StableSwap cUSD/USDCe price with the price on other pools, most users will trade with the reserve StableSwap pool since it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shifts trading volume towards the StableSwap pool.
In other words DEXs have to compete with reserve liquidity. This is a dynamic which we would like to avoid in general. In the current situation we believe that the benefit for the entire Celo ecosystem outweighs the risk of a competitive market intervention.
```

The cUSD/USDCet pool will offer cUSD (and USDCet) at an incorrect price that is either higher (cUSD/USDCet < 1) or lower (cUSD/USDCet > 1) than the market price. The availability of expensive/cheap liquidity can have a price impact that is undesirable for cUSD as it is purely caused by a USDCet price deviation.

 
## Market Interference

> Whenever arbitrageurs have matched cUSD/USDCe prices with the Stable Swap Pool, most users will trade with the reserve Stable Swap Pool. Due to the setup ([see Arbitrage and Dex Competition](#arbitrage-and-dex-competition) ) it has lower fee and slippage especially for larger trades. If DEX router include the Stable Swap Pool, this shift towards Stable Swap Pool will become even more likely.

Reviewer: ```suggestion
Whenever arbitrageurs have matched cUSD/USDCe prices on other pools with the StableSwap pool, most users will trade with the reserve StableSwap pool. Due to the setup (see [Arbitrage and Dex Competition](#arbitrage-and-dex-competition) ) it has lower fees and slippage especially for larger trades. If DEX routers include the StableSwap pool, this shift of trading volume towards the StableSwap pool will become even more likely.
```

> In other words DEXs have to compete with reserve liquidity. This is a dynamic which we would like to avoid in general. In the current situation  we believe that the benefit for the entire Celo ecosystem outweighs the risk of a competitive market intervention:

Reviewer: ```suggestion
In other words DEXs have to compete with reserve liquidity. This is a dynamic which we would like to avoid in general. In the current situation we believe that the benefit for the entire Celo ecosystem outweighs the risk of a competitive market intervention:
```

> 
> 1. We need to enable a larger scale 1:1 solution as far as possible to support cStable adoption.
> 2. The Stable Swap Pool is a temporary setup and interim solution.
> 3. The Stable Swap Pool is only affecting one pair. 
> 4. The setting can be changed. If we notice that with the creation of the Stable Swap Pool, the entire market on Celo becomes distorted, we can pause and propose changing parameters via governance.
> 
> 
> 

Reviewer: ```suggestion

1. We need to enable a larger scale 1:1 solution as far as possible to support cStable adoption.
2. The Stable Swap Pool is a temporary setup and interim solution.
3. The Stable Swap Pool is only affecting one pair. 
4. The setting can be changed. If we notice that with the creation of the Stable Swap Pool, the entire market on Celo becomes distorted, we can pause and propose changing parameters via governance.

```
